### DISCUSSION TITLE
"Open out-of-scope URLs in a default browser" doesn't work whenever any "Domains always allowed to be opened in the PWA browser" are set

### DISCUSSION THREAD

**Discussion by kruns22** [filips123/PWAsForFirefox#197] [General]
When "Domains always allowed to be opened in the PWA browser" is empty, then "Open out-of-scope URLS in a default browser" works fine. When I click on any links that aren't in the scope of the app, they are launched in the default browser.

However, this also breaks my web app. On logon, the app always redirects to a different URL for authentication before redirecting back to the app. "Open out-of-scope URLS in a default browser" breaks this app because it launches the auth URL in the default browser and the PWA window closes itself.

Solution: I add the auth URL to the "Domains always allowed to be opened in the PWA browser". This stops the app from breaking.

However, once any domains are added to "Domains always allowed to be opened in the PWA browser", this seems to break the functionality of "Open out-of-scope URLS in a default browser". External links are now being opened in a new window within the PWA app, not in the default browser, even though "Open out-of-scope URLS in a default browser" is checked.

I'm using Firefox on Windows 11. I've tried with various apps but the one I really want to get working properly is Proton Mail.

Steps to duplicate:
1) In the PWA app, in settings, check "Open out-of-scope URLS in a default browser"
2) Verify that external links are now being launched in the default browser outside the PWA app.
3) In the PWA app, go back to settings, enter a domain under ""Domains always allowed to be opened in the PWA browser". Keep "Open out-of-scope URLS in a default browser" checked.
4) See that external links are now being launched in a new window within the PWA app, rather than in the default browser, even though you kept "Open out-of-scope URLS in a default browser" checked.

Incidentally - fab work on this extension, it's an absolute godsend and this is only a minor issue!

**Comment by filips123** [comment]
External links that are not in "Domains always allowed to be opened in the PWA browser" still open in a default browser for me...

**Comment by mrzoops** [comment]
I am still having this issue. Was there any workaround or fix?

**Reply by filips123** [reply]
How exactly do you have the excluded domains set? Also, please check if this happens in a new web app profile.

**Comment by joao-pedro-braz** [comment]
Same here on a fresh profile.
I have "Open out-of-scope URLs in a default browser (can break some web apps)" set to true, but when I click on a out-of-scope link it always use the web-app instance.
